docs/searching-data.md

```diff
@@ -0,0 +1,71 @@
+
+```py
+client.get_data_series(item_id=327, region_id=1168)
+```
+
+## Lookup contains
+Our ontology is defined in terms of a graph, with metrics/items/regions containing others. In each case, you can see the `'contains'` property in the output of `client.lookup(type, id)`. For example:
+
+```py
+client.lookup('items', 10009)['contains'] 
+```
+
+will return a list of items ids for items that are cereals (item_id = 10009): `[..., 274, 422, ...]`. Once you have those ids, you can use the `client.lookup()` function on each one to find more info, like their names, e.g.: `client.lookup('items', 274)['name']` will return `Corn`.
+
+Similarly, for regions, `client.lookup('regions', 1215)[contains]` will return a list of region ids for regions that are in the US (region_id = 1215): `[13100, 13061, 13053, 13099, ....]`. And each of those can be further looked up e.g. `client.lookup('regions', 13100)` will return `{'name': 'Wisconsin', 'level': 4, 'contains': [139839, 139857, 139863, ...]}`.
+
+## Get descendants
+Using the `lookup()` method, you can get an entity's list of direct children (i.e. country->provinces). However, you may want all of the lower level regions that belong to a higher level region (i.e. country->provinces, districts, coordinates, etc.). To do this, there's a helper function which also gives the option of filtering by region level: [`get_descendant_regions(region_id, descendant_level)`](https://gro-intelligence.github.io/api-client/api.html#api.client.lib.get_descendant_regions).
+
+To look up all descendants of region 1029 (Brazil) that are of level 4 (provinces):
+```py
+from api.client.lib import REGION_LEVELS
+provinces_of_brazil = client.get_descendant_regions(1029, REGION_LEVELS['province'])
+```
+To look up all descendants of region 1029 (Brazil) that are of level 5 (districts): 
+
+```py
+from api.client.lib import REGION_LEVELS
+provinces_of_brazil = client.get_descendant_regions(1029, REGION_LEVELS['district'])
+```
+For more information on region levels, please refer to the [Special properties for regions](./gro-ontology#special-properties-for-regions) section of [Gro Ontology](./gro-ontology.html)
+
+## Lookup belongs
+If you want to find "what entities contain the given entity?" there is a method, [`client.lookup_belongs(type, child_id)`](https://gro-intelligence.github.io/api-client/api.html#api.client.lib.lookup_belongs) that does just that. For example:
+
+```py
+UNITED_STATES = 1215
+client.lookup_belongs('regions', UNITED_STATES)
+```
+will return `[{id: 15, name: 'North America', contains: [1215, 1037, ...], level: 2}, ...]`
```

Comment:
Related to my `search_and_lookup()` comment, "yield" is the more correct term here, rather than "return," since it returns a generator

Reply:
done